.editorconfig

```diff
@@ -0,0 +1,17 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = space
+insert_final_newline = true
+
+[*.jl]
+indent_size = 3
```

Comment:
I guess this does indeed match what the majority of the code does, but it's a bit jarring that it again differs from Oscar.jl and other of our packages...

Since we are currently down to just one other PR. perhaps it is time to think about running JuliaFormatter on the entire package (might make @lkastner happy, too), with whatever settings we prefer (i.e. we'd also need a `JuliaFormatter.toml`)

Reply:
Also fine by me. The same should then be done for AbstractAlgebra.

Reply:
See https://github.com/Nemocas/Nemo.jl/pull/1698 for a proposal.

Reply:
> Also fine by me. The same should then be done for AbstractAlgebra.

Adding the formatting config, yes. With running the formatter on everything I would wait for a moment with only few open relevant PRs